lib/matplotlib/pyplot.py

```diff
@@ -375,25 +406,54 @@ def ioff():
 
     show : show windows (and maybe block)
     pause : show windows, run GUI event loop, and block for a time
+
+    Notes
+    -----
+    If you want the effects of this function to be temporary, it can
+    be used as a context manager, for example::
+
+        with plt.ioff():
+            # interactive mode will be off
+            # figures will not automatically be shown
+            fig1 = plt.figure()
+            # ...
+
+        # This figure will be shown immediately
+        fig2 = plt.figure()
```

Comment:
I would mention the return value in the notes section for the sake of being explicit.

~~~
    The return value is needed to make the function work as a context
    manager. It is not intended to be stored or accessed by the user.
~~~